sf-stream/src/protos/extractor.proto

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+message Transaction {
+  uint64 timestamp = 1;
+  uint64 version = 2;
+  TransactionInfo info = 3;
+  uint64 epoch = 4;
+  uint64 block_height = 5;
+
+  enum TransactionType {
+    GENESIS = 0;
+    BLOCK_METADATA = 1;
+    STATE_CHECKPOINT = 2;
+    USER = 3;
+  }
+  
+  TransactionType type = 6;
+
+  oneof txn_data {
+    BlockMetadataTransaction block_metadata_txn = 7;
+    GenesisTransaction genesis_txn = 8;
+    StateCheckpointTransaction state_checkpoint_txn = 9;
+    UserTransaction user_txn = 10;
+  }
+
+}
+
+message BlockMetadataTransaction {
+  string id = 1;
+  uint64 round = 2;
+  repeated Event events = 3;
+  repeated bool previous_block_votes = 4;
+  string proposer = 5;
+  repeated uint32 failed_proposer_indices = 6;
+}
+
+message GenesisTransaction {
+  WriteSet payload = 1;
+  repeated Event events = 2;
+}
+
+message StateCheckpointTransaction {
+}
+
+message UserTransaction {
+  UserTransactionRequest request = 1;
+  repeated Event events = 2;
+}
+
+message Event {
+  EventKey key = 1;
+  uint64 sequence_number = 2;
+  string type = 3;
+  string data = 4;
+}
+
+message TransactionInfo {
+  string hash = 1;
```

Comment:
All pure hashes and such element should be of `bytes` type, this greatly reduce storage cost on high throughput chains and usually integrates better at the consume side where direct usage of `bytes` can be made.